### DISCUSSION TITLE
Browser Agent Integration with skywalking

### DISCUSSION THREAD

**Discussion by vishalkadam18** [apache/skywalking#12073] [Q&A]
Dear Team,

While integrating Browser Agent (BRUM) into skywalking-ui we are feching this problems???

kindly help to resolve this issue.

Regards,
Vishal Kadam.
![clipboard-1712233227760](https://github.com/apache/skywalking/assets/115125925/7a3247d6-46fa-4ec5-a006-b5a6394ede99)
![clipboard-1712233264671](https://github.com/apache/skywalking/assets/115125925/ac2371ba-012b-4df8-9dc1-78ef33332202)

**Comment by wu-sheng** [comment]
CORS is a common browser side limitation, if you opened TLS.
You should make perf report and your original request request to the same backend domain, then use a Gateway or proxy to forward perf request to OAP.

**Comment by wu-sheng** [comment]
You should learn more about CORS. This is not SkyWalking scope thing.

**Reply by wu-sheng** [reply]
We provided note for you in the client js doc.

https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Comment by niharranjannihar** [comment]
Dear Sheng,
I have setup an Nginx proxy server to resolve the Cors error and successfully resolve the error since I am unable to see the monitoring data in Skywalking, kindly help me with this.

nginx.conf
![image](https://github.com/apache/skywalking/assets/106923369/3f6a8ae0-659e-450a-8e19-b510183ac836)

snippetcode
![image](https://github.com/apache/skywalking/assets/106923369/c189c445-f71e-4811-aa85-0da4bccad0fe)


PFA 

Regards
Nihar

**Reply by wu-sheng** [reply]
Check the storage, such as `service_traffic`. I don't know which storage you are using. If elasticsearch, it is in metric_all_xx index.

**Comment by niharranjannihar** [comment]
I am using the default h2 database, where I can see the data ?

**Reply by wu-sheng** [reply]
You can't see this when you use H2. Pick another option.
